src/utils.rs

```diff
@@ -10,6 +10,69 @@
 
 use syntax::ast::Visibility;
 
+pub trait FindUncommented {
+    fn find_uncommented(&self, pat: &str) -> Option<usize>;
+}
+
+impl FindUncommented for str {
+    fn find_uncommented(&self, pat: &str) -> Option<usize> {
+        let needle = pat.as_bytes();
+        let needle_len = needle.len();
+        let mut match_count: usize = 0;
+        let mut possible_comment = false;
+
+        for (i, b) in self.as_bytes().iter().cloned().enumerate() {
+            if b == needle[match_count] {
+                match_count += 1;
+                if match_count == needle_len {
+                    return Some((i + 1) - needle_len);
+                }
+            } else {
+                match_count = 0;
+            }
+
+            if possible_comment {
+                if b == '/' as u8 {
```

Comment:
I've actually thought about this and it should be alright. Every byte in a multi-byte character has its most significant bit set in UTF-8. ASCII characters we're checking for do not have this byte set. So a byte match is guaranteed to be a character match. I'll add a comment explaining this.

The reason for iterating over `u8` instead of `char` is performance related. The `char` iterator needs to check these bits and doing a copy into 32 bits, whereas the `u8` does not.